hack/portalauth/portalauth.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/gob"
+	"flag"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/gorilla/securecookie"
+	"github.com/gorilla/sessions"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/util/keyvault"
+	utillog "github.com/Azure/ARO-RP/pkg/util/log"
+)
+
+const (
+	SessionName        = "session"
+	SessionKeyExpires  = "expires"
+	SessionKeyUsername = "user_name"
+	SessionKeyGroups   = "groups"
+)
+
+var (
+	username = flag.String("username", "testuser", "username of the portal user")
+	groups   = flag.String("groups", "", "comma-separated list of groups the user is in")
+)
+
+func run(ctx context.Context, log *logrus.Entry) error {
+	flag.Parse()
+
+	_env, err := env.NewCore(ctx, log)
+	if err != nil {
+		return err
+	}
+
+	msiKVAuthorizer, err := _env.NewMSIAuthorizer(env.MSIContextRP, _env.Environment().ResourceIdentifiers.KeyVault)
+	if err != nil {
+		return err
+	}
+
+	portalKeyvaultURI, err := keyvault.URI(_env, env.PortalKeyvaultSuffix)
+	if err != nil {
+		return err
+	}
+
+	portalKeyvault := keyvault.NewManager(msiKVAuthorizer, portalKeyvaultURI)
+
+	sessionKey, err := portalKeyvault.GetBase64Secret(ctx, env.PortalServerSessionKeySecretName, "")
```

Comment:
Is this the same key as used in production?

Reply:
I don't believe so, but these tests have been written to only run in development currently. We would need to figure out a way to run this in production